nixos/modules/services/audio/mpdscribble.nix

```diff
@@ -0,0 +1,231 @@
+        password = "${cfgset.password}"
+        journal = "${cfgset.journal}"
+      '' else "";
+
+  mpdscribbleLastfmConf = mkCfgSection "last.fm" cfg.lastfm;
+  mpdscribbleLibrefmConf = mkCfgSection "libre.fm" cfg.librefm;
+
+  mpdscribbleConf = pkgs.writeText "mpdscribble.conf"
+    (concatStrings
+      mpdscribbleBaseConf
+      mpdscribbleLastfmConf
+      mpdscribbleLibrefmConf);
+in
+{
+  options = {
+    services.mpdscribble = {
+
+      enable = mkOption {
+        default = false;
+        description = ''
+          Whether to enable mpdscribble, a scrobble client for mpd
+        '';
+      };
+
+      user = mkOption {
+        default = "mpdscribble";
+        description = "User account under which mpdscribble runs.";
+      };
+
+      group = mkOption {
+        default = "mpdscribble";
+        description = "Group account under which mpdscribble runs.";
+      };
+
+      mpdHost = mkOption {
+        default = "localhost";
+        example = ''
+          192.168.1.2
+        '';
+        type = type.string;
+        description = ''
+          MPD host
+        '';
+      };
+
+      mpdPort = mkOption {
+        default = 6600;
+        type = type.int;
+        description = ''
+          MPD Port
+        '';
+      };
+
+      logFile = mkOption {
+        default = /dev/null;
+        example = /home/user/.mpdscribble.log;
+        type = type.path;
+        description = ''
+          Path of logfile
+        '';
+      };
+
+      verbosity = mkOption {
+        default = 2;
+        example = 3;
+        type = type.int;
+        description = ''
+          Verbosity of logging.
+          "0" means log only critical errors (e.g. "out of memory");
+          "1"  also logs non-critical  errors (e.g. "server unreachable");
+          "2" logs informational messages (e.g.  "new song");
+          "3" prints  a lot  of debugging messages.
+        '';
+      };
+
+      proxy = mkOption {
+        default = "";
+        example = ''http://your.proxy:8080'';
+        type = type.string;
+        description = ''
+          HTTP Proxy URL.
+        '';
+      };
+
+      lastfm = {
+
+        enable = mkOption {
+          default = false;
+          description = ''
+            Enable Lastfm scrobbling.
+          '';
+        };
+
+        url = mkOption {
+          default = ''http://post.audioscrobbler.com/'';
+          example = ''http://post.audioscrobbler.com/'';
+          type = type.string;
+          description = ''
+            The handshake URL of the scrobbler.
+          '';
+        };
+
+        username = mkOption {
+          default = "";
+          example = "";
+          type = type.string;
+          description = ''
+            Last.fm User
+          '';
+        };
+
+        password = mkOption {
+          default = "";
+          example = "";
+          type = type.;
+          description = ''
+            Last.fm Password
+          '';
+        };
```

Comment:
It wouldn't have to be too complicated, but not sure if that level of flexibility is worth it for a service like this. IMO, all options except those related to scrobble auth and which MPD host to use are excessive (I for one don't care about tweaking where the journal is stored or which user/group the service runs as).